### PR TITLE
test: upload cloud watch CI logs to S3 [DET-3515]

### PR DESCRIPTION
**Pull request by aaron276h**
## Description
We upload the logs directly from cloud watch to S3.

## Test Plan
Ran a full suite of tests here: https://app.circleci.com/pipelines/github/determined-ai/determined/3307/workflows/597e1a9e-04c3-49dd-819a-56c940f997be

Logs from one of the tests from this run can be seen here: https://s3.console.aws.amazon.com/s3/buckets/det-ci-cloud-watch-logs/e2e-gpu-31f7ba5-74285-0/b6f7b1d8-172b-4481-b7e7-e7dcafaa186a/?region=us-west-2&tab=overview

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -447,6 +428,21 @@ commands:
     steps:
       - run: python .circleci/scripts/wait_for_master.py http://<<parameters.host>>:<<parameters.port>>
 
+  upload-cloud-watch-to-s3:
+    parameters:
+      bucket:
+        type: string
+        default: det-ci-cloud-watch-logs
```
Comment (brain-good) [inline .circleci/config.yml:435]: Non-blocking: I usually do the optional arguments after the required

File: .circleci/scripts/upload_cloud_watch_to_s3.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        destination=bucket_name,
+        destinationPrefix=prefix
+    )["taskId"]
+
+    print(f"Uploading logs to s3://{bucket_name}/{prefix}/{task_id}.")
+    return task_id
+
+
+def wait_for_export_task(task_id: str) -> None:
+    while True:
```
Comment (brain-good) [inline .circleci/scripts/upload_cloud_watch_to_s3.py:24]: blocking: let's not loop forever -- this is bounded to 5 hours but that's still a long time

File: .circleci/config.yml (hunk 1 above)
Comment (aaron276h) [reply .circleci/config.yml:435]: done

File: .circleci/scripts/upload_cloud_watch_to_s3.py (hunk 2 above)
Comment (aaron276h) [reply .circleci/scripts/upload_cloud_watch_to_s3.py:24]: Good idea, added a 5 min timeout for this.

**Pull request merged**